### ISSUE TITLE
[YSQL] Batch nested loops closing connection when querying system catalog tables for Hasura

### ISSUE THREAD

**Issue by mislam-yb** [yugabyte/yugabyte-db#16025]
Jira Link: [DB-5428](https://yugabyte.atlassian.net/browse/DB-5428)
### Description

When using batched nested loops the following query causes the connection to disconnect:

```
-- enable BNL
set yb_bnl_batch_size = 1024;

SELECT
    COALESCE(
        json_agg(Row_to_json(functions)),
        '[]' :: JSON
    )
from
    (
        SELECT
            *
        FROM
            (
                SELECT
                    p.proname :: text AS function_name,
                    pn.nspname :: text AS function_schema,
                    pd.description,
                    CASE
                        WHEN p.provariadic = 0 :: oid THEN false
                        ELSE true
                    END AS has_variadic,
                    CASE
                        WHEN p.provolatile :: text = 'i' :: character(1) :: text THEN 'IMMUTABLE' :: text
                        WHEN p.provolatile :: text = 's' :: character(1) :: text THEN 'STABLE' :: text
                        WHEN p.provolatile :: text = 'v' :: character(1) :: text THEN 'VOLATILE' :: text
                        ELSE NULL :: text
                    END AS function_type,
                    pg_get_functiondef(p.oid) AS function_definition,
                    rtn.nspname :: text AS return_type_schema,
                    rt.typname :: text AS return_type_name,
                    rt.typtype :: text AS return_type_type,
                    obj_description(p.oid) AS comment,
                    p.proretset AS returns_set,
                    (
                        SELECT
                            COALESCE(
                                json_agg(
                                    json_build_object(
                                        'schema',
                                        q.schema,
                                        'name',
                                        q.name,
                                        'type',
                                        q.type
                                    )
                                ),
                                '[]' :: json
                            ) AS "coalesce"
                        FROM
                            (
                                SELECT
                                    pt.typname AS name,
                                    pns.nspname AS schema,
                                    pt.typtype AS type,
                                    pat.ordinality
                                FROM
                                    unnest(
                                        COALESCE(p.proallargtypes, p.proargtypes :: oid [])
                                    ) WITH ORDINALITY pat(oid, ordinality)
                                    LEFT JOIN pg_type pt ON pt.oid = pat.oid
                                    LEFT JOIN pg_namespace pns ON pt.typnamespace = pns.oid
                                ORDER BY
                                    pat.ordinality
                            ) q
                    ) AS input_arg_types,
                    to_json(COALESCE(p.proargnames, ARRAY [] :: text [])) AS input_arg_names,
                    p.pronargdefaults AS default_args,
                    p.oid :: integer AS function_oid
                FROM
                    pg_proc p
                    JOIN pg_namespace pn ON pn.oid = p.pronamespace
                    JOIN pg_type rt ON rt.oid = p.prorettype
                    JOIN pg_namespace rtn ON rtn.oid = rt.typnamespace
                    LEFT JOIN pg_description pd ON p.oid = pd.objoid
                WHERE
                    pn.nspname :: text !~~ 'pg_%' :: text
                    AND(
                        pn.nspname :: text <> ALL (ARRAY ['information_schema'::text])
                    )
                    AND NOT(
                        EXISTS (
                            SELECT
                                1
                            FROM
                                pg_aggregate
                            WHERE
                                pg_aggregate.aggfnoid :: oid = p.oid
                        )
                    )
            ) as info -- WHERE function_schema='schema'
        WHERE
            function_schema = 'schema'
            AND has_variadic = FALSE
            AND return_type_type = 'c'
        ORDER BY
            function_name ASC
    ) as functions;
```

This error occured for the latest debug build for 2.17.2.0, and was reproduced by @tanujnay112.

[DB-5428]: https://yugabyte.atlassian.net/browse/DB-5428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by tverona1** [comment]
Error is: ERROR:  cache lookup failed for type 1

**Comment by tanujnay112** [comment]
Fixed by 5d4d88fb7b65cde02d5de9c07ece0c1a22b378a4
